client/capstone-app/src/pages/MapPage.jsx

```diff
@@ -0,0 +1,138 @@
+import React, { useEffect, useState } from "react";
+import MapWithRoute from '../components/MapWithRoute'
+import { useParams } from "react-router-dom";
+
+
+const MapPage = () => {
+    const [directions, setDirections] = useState(null)
+    const [currentRadius, setCurrentRadius] = useState(null)
+    const [destinationRadius, setDestinationRadius] = useState(null)
+    const {currentCoordinates, destinationCoordinates} = useParams()
+    const [error, setError] = useState(null)
+
+    function convertCoordinates(coordString) {
+      return coordString.split(',').map(Number);
+    }
+
+    const fetchDirections = async () => {
+        try {
+        const response = await fetch(`https://api.openrouteservice.org/v2/directions/driving-car?api_key=5b3ce3597851110001cf6248d2ab3de842a1474a865cff3fff2e2787&start=${currentCoordinates}&end=${destinationCoordinates}`, {
+          method: 'GET',
+          headers: {
+            'Accept': 'application/json, application/geo+json, application/gpx+xml, img/png; charset=utf-8'
+          },
+        });
+        const data = await response.json();
+        setDirections(data)
+      } catch (error) {
+        console.error(error);
+      }
+    }
+
+    const currentCoordArray = convertCoordinates(currentCoordinates)
+    const destCoordArray = convertCoordinates(destinationCoordinates)
+
+  const fetchCurrentRadius = async () => {
+    try {
+      const response = await fetch('https://api.openrouteservice.org/v2/isochrones/foot-walking', {
+      method: 'POST',
+      headers: {
+        'Accept': 'application/json, application/geo+json, application/gpx+xml, img/png; charset=utf-8',
+        'Authorization': '5b3ce3597851110001cf6248d2ab3de842a1474a865cff3fff2e2787',
```

Comment:
Put into var. THis is a string repeated twice so apply DRY principle. -->https://en.wikipedia.org/wiki/Don%27t_repeat_yourself